lib/dynflow/action.rb

```diff
@@ -584,6 +590,18 @@ def self.singleton_locked?(world)
         false
       end
     end
+
+    def update_step_telemetry(phase)
+      Dynflow::Telemetry.with_instance do |t|
+        t.observe_histogram(:dynflow_step_real_time,
+                            @step.real_time,
+                            :action => self.class.to_s, :phase => phase)
+        t.observe_histogram(:dynflow_step_execution_time,
+                            @step.execution_time,
```

Comment:
When I looked at the collected data in Prometheus, the histograms looked weird, but maybe I was just interpreting it wrong. I'll need to wrap my head around the whole buckets thing so this may need tweaking.

Reply:
There was a bug fixed in core recently, units were incorrect: https://github.com/theforeman/foreman/pull/5874/files

Histogram can't be easily plotted in time, it's better to look on them aggregated https://www.google.cz/search?q=histogram

But if you need to look thorugh the time, it's usually better just to pick several buckets and draw them together as line graph. Prometheus can calculate approximate percentiles which is usually very good way of interpreting data (99th, 95th percentile): https://kausal.co/blog/prometheus-histograms-multiple-queries/